packages/app/src/hooks/useRowSidePanel.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import LZString from 'lz-string';
+import { createParser, useQueryState } from 'nuqs';
+import { ChartConfigWithDateRange } from '@hyperdx/common-utils/dist/types';
+
+import DBRowSidePanel from '@/components/DBRowSidePanel';
+import { useSources } from '@/source';
+
+type SidePanelState = {
+  sourceId: string;
+  rowWhere: string;
+  tab?: string;
+  dbSqlRowTableConfig?: ChartConfigWithDateRange;
+};
+
+type SidePanelStack = SidePanelState[];
+
+// Compress and decompress the value to save space in the URL
+const deflateParser = createParser({
+  parse: (value: string) => {
+    return JSON.parse(LZString.decompressFromBase64(value)) as SidePanelStack;
```

Comment:
I think we should keep them user readable, that way it's easy to interact programmatically as well (people who build URLs dynamically from retool and other internal platforms for example)